### ISSUE TITLE
[Windows] Some icon fonts not working in unpacked apps

### ISSUE THREAD

**Issue by GuidoNeele** [dotnet/maui#21993]
### Description

**Some** icon fonts are not working in unpacked apps when using them with FontImageSource. They do work when they are used with labels.

The icon fonts I tested:
- Font awesome (doesn't work)
- ionicons (works)
- MaterialIcons (doesn't work)

![image](https://github.com/dotnet/maui/assets/1301586/74fdf808-79ad-40cd-b948-ecf530506a53)

```xaml
<Button x:Name="CounterBtn"
        Clicked="OnCounterClicked"
        FontFamily="OpenSansSemibold"
        HorizontalOptions="Fill"
        ImageSource="{FontImage &#xf1b9;,
                                FontFamily=FontAwesome,
                                Color=White}"
        SemanticProperties.Hint="Counts the number of times you click"
        Text="Click me" />

<Button Clicked="OnCounterClicked"
        FontFamily="OpenSansSemibold"
        HorizontalOptions="Fill"
        ImageSource="{FontImage &#xf30c;,
                                FontFamily=ionicons,
                                Size=44}"
        SemanticProperties.Hint="Counts the number of times you click"
        Text="Click me" />

<Button Clicked="OnCounterClicked"
        FontFamily="OpenSansSemibold"
        HorizontalOptions="Fill"
        ImageSource="{FontImage Glyph=&#xe000;,
                                FontFamily=MaterialIcons,
                                Size=44}"
        SemanticProperties.Hint="Counts the number of times you click"
        Text="Click me" />
```

### Steps to Reproduce

1. Download attached reproduction repository
2. Open solution
3. Build SharedLib as Release
4. Build MauiIconTest as Release
5. Go to <repository_path>\MauiIconTest\bin\Release\net8.0-windows10.0.19041.0\win10-x64\
6. Start MauiIconTest.exe

OR

1. Start a new MAUI project
2. Add FontAwesome font to Resources/Fonts directory
3. Add font to MauiProgram.cs
```csharp
.ConfigureFonts(fonts =>
{
    fonts.AddFont("FontAwesome.ttf", "FontAwesome");
    fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
    fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
});
```
4. Add the following lines to csproj
```xml
<WindowsPackageType>None</WindowsPackageType>
<WindowsAppSDKSelfContained>true</WindowsAppSDKSelfContained>
<WindowsAppSdkDeploymentManagerInitialize>false</WindowsAppSdkDeploymentManagerInitialize>
```
5. Add button to MainPage
```xaml
<Button Clicked="OnCounterClicked"
        HorizontalOptions="Fill"
        ImageSource="{FontImage &#xf30c;,
                                FontFamily=FontAwesome,
                                Size=44}"
        SemanticProperties.Hint="Counts the number of times you click"
        Text="FontAwesome" />
```
6. Build application and run it.

### Link to public reproduction project repository

https://github.com/GuidoNeele/MauiIconTest

### Version with bug

8.0.20 SR4

### Is this a regression from previous behavior?

No, this is something new, Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

net8.0-windows10.0.19041.0

### Did you find any workaround?

No didn't find a workaround.
A workaround could be to use a different icon font than FontAwesome because ionicons does work.

**Comment by RoiChen001** [comment]
Can repro this issue at Windows platform on the latest 17.10 Preview 4(8.0.20)

**Comment by mattleibow** [comment]
I see material icons is a `.otf` file, so that may not be supported - but I need to confirm. Was there anything in the logs?

I wonder if I have an issue in my targets file that is preventing something - I will have a look.

**Comment by mattleibow** [comment]
OK, I managed to repro this, I have no idea why it is failing as there are no errors in the logs:

[MauiApp4.zip](https://github.com/dotnet/maui/files/15083064/MauiApp4.zip)

I will; have a look and try debug.

@rmarinho I think I recall you saying that you had an issue with FontAwesome in your app?

**Comment by mattleibow** [comment]
There are no errors in maui, so for some reason Win2D just does not draw the right thing. I'll log an issue with that team.

**Comment by BurkusCat** [comment]
I'm fairly sure this is essentially the same issue that #21333 was except the title is more accurate in this issue here. I've had the same issue with `materialdesignicons-webfont.ttf` and I had hoped #21144 would fix the issue for me but it didn't. I tried converting the font into a few different font file formats but nothing worked. After PR #21144 I don't get any errors any more when running the app and wasn't able to find any useful logs to hint what might be going wrong.

**Comment by mattleibow** [comment]
I figured out the issue. The font name had spaces and the Uri was encoding things.

**Comment by GuidoNeele** [comment]
Just tried the nightly and fix #22646 indeed solved this issue.
Thanks @mattleibow!

**Comment by BurkusCat** [comment]
8.0.60 this appears to be fixed for me with material design icons. Thank you :)
